src/Component/src/Metadata/MetadataInterface.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Metadata;
+
+/**
+ * @method string|null getStateMachineComponent()
+ */
+interface MetadataInterface
+{
+    public function getAlias(): string;
+
+    public function getApplicationName(): string;
+
+    public function getName(): string;
+
+    public function getHumanizedName(): string;
+
+    public function getPluralName(): string;
+
+    public function getDriver(): string|false;
+
+    /**
+     * @return ?string
+     */
+    public function getTemplatesNamespace(): ?string;
+
+    /**
+     * @return string|array
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function getParameter(string $name);
+
+    /**
+     * @return class-string $name
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function getClass(string $name): string;
+
+    /**
+     * Return all the metadata parameters.
+     */
+    public function getParameters(): array;
+
+    public function hasParameter(string $name): bool;
+
+    public function hasClass(string $name): bool;
```

Comment:
Would be cool to sort methods list the same order as in class.

right now we have
```php
getClass
getParameters
hasParameter
hasClass
```